src/pages/images/actions/forms/UploadImageForm.tsx

```diff
@@ -0,0 +1,181 @@
+import { ChangeEvent, FC, useState } from "react";
+import { useEventQueue } from "context/eventQueue";
+import { useFormik } from "formik";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { createImageAlias, createImageFromTarball } from "api/images";
+import {
+  ActionButton,
+  Button,
+  Form,
+  Input,
+  Modal,
+} from "@canonical/react-components";
+import * as Yup from "yup";
+import { Link } from "react-router-dom";
+import { humanFileSize } from "util/helpers";
+import ProgressBar from "components/ProgressBar";
+import { UploadState } from "types/storage";
+import Loader from "components/Loader";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+
+interface Props {
+  close: () => void;
+  project: string;
+}
+
+const UploadImageForm: FC<Props> = ({ close, project }) => {
+  const eventQueue = useEventQueue();
+  const toastNotify = useToastNotification();
+  const [uploadState, setUploadState] = useState<UploadState | null>(null);
+  const queryClient = useQueryClient();
+
+  const notifySuccess = () => {
+    const uploaded = <Link to={`/ui/project/${project}/images`}>uploaded</Link>;
+    toastNotify.success(<>Image {uploaded}.</>);
+  };
+
+  const changeFile = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files) {
+      void formik.setFieldValue("fileList", e.target.files);
+    }
+  };
+
+  const getImageToTarballBody = (fileList: FileList): File | FormData => {
```

Comment:
```suggestion
  const getImageUploadBody = (fileList: FileList): File | FormData => {
```
Tarball seems wrong, as this is also handling the case of root image plus metadata file, that can be gzip, without a tarball.